cpp/src/strings/regex/regcomp.cpp

```diff
@@ -570,182 +570,180 @@ class regex_parser {
  * @brief The compiler converts class list into instructions.
  */
 class regex_compiler {
-  reprog& m_prog;
-
-  struct Node {
+  struct and_node {
     int id_first;
     int id_last;
   };
 
-  int cursubid;
-  int pushsubid;
-  std::vector<Node> andstack;
-
-  struct Ator {
+  struct re_operator {
     int t;
     int subid;
   };
 
-  std::vector<Ator> atorstack;
-
-  bool lastwasand;
-  int nbra;
+  reprog& _prog;
+  std::stack<and_node> _and_stack;
+  std::stack<re_operator> _operator_stack;
+  bool _last_was_and;
+  int _bracket_count;
+  regex_flags _flags;
 
-  regex_flags flags;
+  inline void push_and(int first, int last) { _and_stack.push({first, last}); }
 
-  char32_t yy;
-  int yyclass_id;
-
-  inline void pushand(int f, int l) { andstack.push_back({f, l}); }
-
-  inline Node popand(int op)
+  inline and_node pop_and()
   {
-    if (andstack.size() < 1) {
-      // missing operand for op
-      int inst_id = m_prog.add_inst(NOP);
-      pushand(inst_id, inst_id);
+    if (_and_stack.empty()) {
+      auto const inst_id = _prog.add_inst(NOP);
+      push_and(inst_id, inst_id);
     }
-    Node node = andstack[andstack.size() - 1];
-    andstack.pop_back();
+    auto const node = _and_stack.top();
+    _and_stack.pop();
     return node;
   }
 
-  inline void pushator(int t)
+  inline void push_operator(int token, int subid = 0)
   {
-    Ator ator;
-    ator.t     = t;
-    ator.subid = pushsubid;
-    atorstack.push_back(ator);
+    _operator_stack.push(re_operator{token, subid});
   }
 
-  inline Ator popator()
+  inline re_operator const pop_operator()
   {
-    Ator ator = atorstack[atorstack.size() - 1];
-    atorstack.pop_back();
+    auto const ator = _operator_stack.top();
```

Comment:
Can we rename all `ator` variables to `operator` instead to match the actual class's rename?

Reply:
Unfortunately `operator` is a reserved word. 
I felt nostalgic for `ator` after all of this time and thought it harmless for a variable name in a 3-line function.

Reply:
Good point, forgot about the keyword. I would agree that it's harmless in just one small function, however I was also hoping to replace it in the subsequent `eval_until` function, which is much longer. I like that we use the same name consistently in both, it reinforces that they're the same type of object. It would be nice to use a more descriptive name though. Unfortunately `re_operator` (the name of the type) is also already taken. One pattern that I've seen used elsewhere to avoid clashing with reserved keywords is just adding a trailing underscore e.g. `operator_`, and I think that would be more informative than `ator`. I won't block this PR on the rename request, though.